packages/edit-site/src/components/global-styles/ui.js

```diff
@@ -231,7 +233,12 @@ function GlobalStylesBlockLink() {
 		if ( newPath !== currentPath ) {
 			navigator.goTo( newPath, { skipFocus: true } );
 		}
-	}, [ selectedBlockClientId, selectedBlockName, blockHasGlobalStyles ] );
+	}, [
+		selectedBlockClientId,
+		selectedBlockName,
+		blockHasGlobalStyles,
+		navigator,
```

Comment:
This new dependency on `navigator` is causing the `push-to-global-styles` test to fail. The test expects to need to click the "Headings" button in the Styles panel, but the Styles panel path is now updated whenever the selected block changes, which means the Headings panel is opened automatically when a Heading block is selected.

This saves a click, but I'm not sure what the intended behaviour is. Either:
1. The Styles panel path should update whenever the selected block changes - saves on clicks, but is different to the current behaviour.
2. The Styles panel path does not change regardless of the selected block, and it will only change with direct interaction (i.e. navigation) within the Styles panel.

If it's number 1 we should update the test to remove these additional navigation clicks; if it's number 2 then I think we should remove the dependency on `navigator` here.

Reply:
Curiously I don't remember adding those lines. I reverted the `navigator` dependency. The linter is not complaining and the tests are passing, so I think it's ok this way. Thanks for the research about this!